connector.go

```diff
@@ -36,11 +36,18 @@ func (c *connector) Connect(ctx context.Context) (driver.Conn, error) {
 	dialsLock.RLock()
 	dial, ok := dials[mc.cfg.Net]
 	dialsLock.RUnlock()
+
 	if ok {
-		mc.netConn, err = dial(ctx, mc.cfg.Addr)
+		dctx := ctx
+		if mc.cfg.Timeout > 0 {
+			var cancel context.CancelFunc
+			dctx, cancel = context.WithTimeout(ctx, c.cfg.Timeout)
+			defer cancel()
```

Comment:
`defer cancel()` is not needed.

Reply:
this avoids context leak, though it's not nesseary here
but code like `dctx, _ = context.WithTimeout()` ,  not the good style.

Reply:
ok.